GoogleUtilities/README.md

```diff
@@ -15,15 +15,23 @@ Instructions on how to adopt the app delegate swizzler for use by your SDK are a
 
 ## Development
 
-Follow the subsequent instructions to develop, debug, and unit test
-GoogleUtilities:
-
-```
-$ git clone git@github.com:firebase/firebase-ios-sdk.git
-$ cd firebase-ios-sdk/GoogleUtilities/Example
-$ pod update
-$ open GoogleUtilities.xcworkspace
-```
+Follow the subsequent instructions to develop, debug, unit test, and
+integration test FirebaseFunctions:
+
+### Prereqs
+
+- At least CocoaPods 1.6.0
+- Install [cocoapods-generate](https://github.com/square/cocoapods-generate)
```

Comment:
Should we also recommend using Bundler to manage all required Gems? E.g.
`bundle install`
then
`bundle exec pod gen GoogleUtilities.podspec`

Reply:
Do end users typically do this kind of thing, or is this more typically for CI?

I worry that while this is more complete it adds more choice to the instructions, and having to choose is just one more way to get things wrong.

Also, if we do recommend bundle install, I think we should recommend something local, rather than the rubygems default location. Something like:

```sh
bundle install --path vendor/bundle
```

Reply:
@wilhuff Unfortunately, I don't have a representable data on usage. From my recent out-of-Google experience using `rvm` + `bundler` was a default setup for most of iOS projects I was working with. 

In this case Gems versions depend on `Gemfile` even without pre-appending with `bundle exec` which is pretty convenient. Relying on bundler enables developers to have a more consistent environment (at least Ruby part), so there is no need to install/update individual Gems for each project. I would refer to `rvm` + `bundler` as a recommended way.

As for `--path vendor/bundle` it can be an option, though I don't remember having any issues with default location.

Reply:
Let's take this discussion and solve this particular problem in #2912.  `pod gen` is for more that CI. It is intended to be a requirement for development in this repo.  Nothing about the Gemfile should matter to end users, right?

Reply:
I don't think Firebase users need to care of Gemfile - it's for contributors only.

Reply:
Correct, end users don't care about our Gemfile, but contributors in the repo do care.

The thing I'm bringing up here is that it's hard to bring up new team members on code in this repo without docs that clearly articulate a happy path and have an opinion on how we work. I'm fine with adding additional documentation for nonstandard paths, but I think that as much as we can, this should be isolated or called out specifically as an alternate.

To date we haven't been requiring everyone to use rvm + bundler and those aren't in our docs for how to contribute so I'd suggest that this PR doesn't introduce that requirement indirectly here.

Reply:
@wilhuff  I agree that an additional set of tools like introduces additional steps for a new team member, so in a general case it is better to avoid anything extra.

As for  rvm + bundler in particular, I still think, we should consider using them because:
- it allows to define Ruby environment based on the source code, so developers can be sure that a set of commands works in the same way everywhere
- it is kind of a standard way to work with ruby and gems

Reply:
I don't disagree with you.

The tension is between the goal of completely reproducible development environments like what you get with rvm and developer friendliness to newcomers that might want to contribute that aren't otherwise professional developers.

In particular, every setup step we add is a place for developers to fail, and every bit of automation we introduce makes these things our problems when they fail.

I'm not arguing strongly for either side, just trying to highlight how painful it was recently to spin up another member of the Firestore team.

I grant that not many people are going to be mucking about with GoogleUtilities specifically so we can even skip resolving this here and just go with whatever we would recommend without an alternate. The question I'm asking is largely about how we develop the rest of pods in this repo.

Reply:
@wilhuff So what would you recommend?  Document a using rvm and a not-using rvm set of instructions? And how would they be positioned so someone who just wants to get started goes down the happy path?

Reply:
I suggest that we have a single doc for getting your environment with a few alternatives. Something like the way the [cocoapods getting started docs](https://guides.cocoapods.org/using/getting-started.html) work.

Our goal for the alternatives should be that each one results in end users being able to run the same commands. That is, no alternative is valid if you end up having to run `bundle exec` as a prefix of every command.

Then each document like this one can just refer to the get your environment setup doc and give the commands that work either way.

This would avoid the burden of multiple sets of instructions per pod we're documenting. WDYT?

Reply:
Docs will be updated once all pods have `pod gen` set up and we start removing the old xcodeproj dev environment.